.hooks/pre-commit

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+FORBIDDEN=( private_key )
+
+for i in "${FORBIDDEN[@]}"
+do
+  git diff --cached --name-only | \
+        GREP_COLOR='4;5;37;41' xargs grep --color --with-filename -n ${i} --exclude .hooks/pre-commit && \
+        echo 'COMMIT REJECTED: Found "'${i}'" references. Please remove them before committing.' && exit 1
+done
+
+exit 0
```

Comment:
FYI: The docs suggest it's possible to write hooks in any language, so if maintaining a shell script ever becomes a pain, we could rewrite it in ruby or node or something.